text/0000-prelude-2021.md

```diff
@@ -0,0 +1,164 @@
+- Feature Name: `prelude_2021`
+- Start Date: 2021-02-16
+- RFC PR: [rust-lang/rfcs#0000](https://github.com/rust-lang/rfcs/pull/0000)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+[summary]: #summary
+
+A new prelude for the 2021 edition, featuring several extra traits.
+
+# Motivation
+[motivation]: #motivation
+
+While types and free functions can be added to the prelude independent of edition boundaries, the same is not true for traits. Adding a trait to the prelude can cause compatibility issues because calls to methods named the same as methods of the newly in scope traits can become ambiguous. Because editions are opt-in, they represent an opportunity to add new traits to the prelude.
```

Comment:
> Adding a trait to the prelude can cause compatibility issues

This is true, but AFAIK T-libs doesn't actually consider that a breaking change (since otherwise every time they added an inherent method, or a new method to a trait, it could break code). Not sure if you want to mention that or not.

Reply:
I think the official terminology is that it's a "minor change": https://rust-lang.github.io/rfcs/1105-api-evolution.html#minor-change-adding-a-defaulted-item

So allowed despite the breakage.

Reply:
I think it's still worth keeping this explanation, and just expanding it to make it clear that while we *can* make such a change, we may choose not to due to the possibility of breakage.

The explanation should also make clear that many traits we might want to add are especially likely to cause breakage, because they may conflict with similar traits from the crate ecosystem that exist to fill the same need that the standard library trait would.